pkg/csi/service/common/placementengine/placement.go

```diff
@@ -246,6 +246,65 @@ func getExpandedTopologySegments(ctx context.Context, requestedSegments map[stri
 	return completeTopologySegments, nil
 }
 
+// GetAllAccessibleDSInTopology returns a list of accessible datastores for requested topology.
+func GetAllAccessibleDSInTopology(ctx context.Context,
+	topologyArr []map[string]string, vcenter *cnsvsphere.VirtualCenter) ([]*cnsvsphere.DatastoreInfo,
+	error) {
+	log := logger.GetLogger(ctx)
+	nodeMgr := node.GetManager(ctx)
+
+	var candidateDatastores []*cnsvsphere.DatastoreInfo
+	for _, reqSegment := range topologyArr {
+		// Fetch the complete hierarchy of topology segments.
+		completeTopologySegments, err := getExpandedTopologySegments(ctx, reqSegment, nodeMgr)
```

Comment:
Can you remove the `getExpandedTopologySegments` part  (Line: 258 - 271) , and directly use 
`reqSegment..GetSegments()` in `GetHostsForSegment` ?

Reply:
Done